lib/src/common/disposable.dart

```diff
@@ -26,20 +26,41 @@ abstract class _Disposable {
   Future<Null> dispose();
 }
 
-class InternalDisposable implements _Disposable {
+class InternalDisposable implements _Disposable, ManagedDisposer {
```

Comment:
Yep, I thought about that briefly. `ManagedDisposer` seemed a little bit like something a consumer shouldn't be able to construct, but that's not really a big deal. I'll go ahead and do that.